### PR TITLE
Merging the cps class based off provided examples

### PR DESCRIPTION
**Pull request by mattp0**
Tests gathered from other groups and macros from Dr Hartman

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    Catch::Session testing;
+    int catchresult;
+    catchresult = testing.applyCommandLine(argc, argv);
+    if (!catchresult) {
+        std::cout << "Begin tests for class shapes and its derived classes: \n";
+        catchresult = testing.run();
+        std::cout << "End of automated tests.\n"; 
+    }
+    return 0;
+}
```
Comment (mckadesorensen) [inline main.cpp:19]: Not something to be fixed but new lines at the end of a file are good practice for whatever reason (I should look up why)

**Review by mckadesorensen — APPROVED** [review]

**Review by TailonR — APPROVED** [review]
Looks Good.

File: main.cpp (hunk 1 above)
Comment (mattp0) [reply main.cpp:19]: like return 0; newline newline?

Comment (mckadesorensen) [reply main.cpp:19]: Screen Shot 2020-04-01 at 3 06 48 PM

Comment (mckadesorensen) [reply main.cpp:19]: No after the last bracket in this case.

**Pull request merged**